Python/specialize.c

```diff
@@ -2372,14 +2372,159 @@ binary_op_fail_kind(int oparg, PyObject *lhs, PyObject *rhs)
 }
 #endif
 
+/** Binary Op Specialization Extensions */
+
+/* float-long */
+
+static int
+float_long_guard(PyObject *lhs, PyObject *rhs)
```

Comment:
Maybe
```suggestion
float_compact_long_guard(PyObject *lhs, PyObject *rhs)
```
or
```suggestion
float_compact_int_guard(PyObject *lhs, PyObject *rhs)
```
or
```suggestion
pyfloat_pylong_guard(PyObject *lhs, PyObject *rhs)
```
It would make the name longer (and to be consistent, the name of several other methods), but would describe better what the guard is.

In bytecodes.c currently INT is used (e.g. `BINARY_ADD_INT`) for an (exact) Python int. Although not always: `BINARY_SUBSCR_LIST_INT` deopts if the index is not compact.

The current name is the shortest, so maybe stick with that after all.

Reply:
I'll go with `float_compactlong_guard`.